### PR TITLE
Enable setting encyrption properties for volumes and filesystems

### PR DESCRIPTION
**Pull request by sdwilsh**
## Description

This isn't useful if you want to setup encryption with a prompt, but if
you have a local file with the key, this allows one to use this role to
setup a volume or filesystem with encryption enabled.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

### CODE REVIEW

File: tasks/manage_zfs.yml [hunk 1]
```diff
@@ -90,15 +90,18 @@
...
       logbias: "{{ item.logbias|default(omit) }}"
       mountpoint: "{{ item.mountpoint|default(omit) }}"
       primarycache: "{{ item.primarycache|default(omit) }}"
       quota: "{{ item.quota|default(omit) }}"
       recordsize: "{{ item.recordsize|default(omit) }}"
       sharenfs: "{{ item.sharenfs|default(omit) }}"
-      sync: "{{ item.sync|default(omit) }}"
       snapdev: "{{ item.snapdev|default('hidden')}}"
       snapdir: "{{ item.snapdir|default('hidden')}}"
+      sync: "{{ item.sync|default(omit) }}"
```
Comment (sdwilsh) [inline tasks/manage_zfs.yml:104]: I moved some stuff around here and below because these keys were very nearly alphabetized.

Comment (mrlesmithjr) [reply tasks/manage_zfs.yml:104]: I definitely appreciate that ;)

File: tasks/manage_zfs.yml [hunk 2]
```diff
@@ -90,15 +90,18 @@
       atime: "{{ item.atime|default(omit) }}"
       compression: "{{ item.compression|default(omit) }}"
       dedup: "{{ item.dedup|default(omit) }}"
+      encryption: "{{ item.encryption|default(omit) }}"
```
Comment (mrlesmithjr) [inline tasks/manage_zfs.yml:93]: Need to add examples of these options in defaults/main.yml so they are clear

**Review by mrlesmithjr — CHANGES_REQUESTED** [review]

File: tasks/manage_zfs.yml (hunk 2 above)
Comment (sdwilsh) [reply tasks/manage_zfs.yml:93]: Ironic that I missed that, given that I just did a PR to add docs from a different PR!

**Comment by sdwilsh** [comment]
Anything else you need from me on this one, @mrlesmithjr?

**Review by mrlesmithjr — APPROVED** [review]

**Pull request merged**